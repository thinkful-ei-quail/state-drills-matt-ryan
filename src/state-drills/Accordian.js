import React from 'react';

class Accordian extends React.Component {
  static defaultProps = {
    sections: []
  }

  state = {
    activeTabIndex: null
  }

  handleActiveTab (index) {
      this.setState({activeTabIndex: index});
  }

  renderTab(section, tabId, activeTabIndex) {
    return (
      <li key={tabId}>
        <button onClick={() => this.handleActiveTab(tabId)}>
          {section.title}
        </button>
        {(activeTabIndex === tabId) && <p>{section.content}</p>}
      </li>
    );
  }

  render () {
    return (
      <ul>
        {this.props.sections.map((section, tabId) => {
          return this.renderTab(section, tabId, this.state.activeTabIndex);
        })}
      </ul>
    );
  }
}

export default Accordian;