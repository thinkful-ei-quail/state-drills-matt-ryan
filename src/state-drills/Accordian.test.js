import React from 'react';
import ReactDOM  from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordian from './Accordian';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('This is the Accordian test.', () => {
  const sectionsProps = [
    {
      title: 'Section One',
      content: 'bla bla bla'
    },
    {
      title: 'Section One',
      content: 'bla bla bla'
    },
    {
      title: 'Section One',
      content: 'bla bla bla'
    }
  ]
  it('Renders empty <li></li> given no accordian', () => {
    const wrapper = shallow(<Accordian />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('Renders no section active by default', () => {
    const wrapper = shallow(<Accordian sections={sectionsProps}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Opens a clicked section', () => {
    const wrapper = shallow(<Accordian sections={sectionsProps}/>);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Only have one section open at a time', () => {
    const wrapper = shallow(<Accordian sections={sectionsProps}/>);
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});