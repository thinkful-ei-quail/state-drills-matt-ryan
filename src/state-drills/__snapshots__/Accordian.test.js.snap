// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is the Accordian test. Only have one section open at a time 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
    <p>
      bla bla bla
    </p>
  </li>
</ul>
`;

exports[`This is the Accordian test. Opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
    <p>
      bla bla bla
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
</ul>
`;

exports[`This is the Accordian test. Renders empty <li></li> given no accordian 1`] = `<ul />`;

exports[`This is the Accordian test. Renders no section active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section One
    </button>
  </li>
</ul>
`;
